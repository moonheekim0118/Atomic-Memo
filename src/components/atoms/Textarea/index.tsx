import React, { useRef, useEffect } from 'react';
import styled from 'styled-components';

interface Props {
  value?: string;
  onChange?: (e: React.MouseEvent) => void;
  size: 'main' | 'title';
}

interface optionType {
  rows: number;
  fontSize: string;
  placeholder: string;
}

const getOptions = (size: 'main' | 'title'): optionType => {
  if (size === 'main') {
    return { rows: 10, fontSize: '1.2rem', placeholder: 'Contents' };
  }
  return { rows: 2, fontSize: '2rem', placeholder: 'Title' };
};

const Textarea = ({ value, onChange, size }: Props) => {
  const textInput = useRef(null);
  const stylingProps = getOptions(size);
  useEffect(() => {
    // 렌더링시 자동으로 focus
    textInput.current.focus();
  }, []);
  return (
    <StyledTextArea
      {...stylingProps}
      value={value}
      onChange={onChange}
      ref={textInput}
    />
  );
};

const StyledTextArea = styled.textarea<optionType>`
  border: none;
  font-size: ${(props) => props.fontSize};
  width: 100%;
  overflow: hidden;
  resize: none;

  &:focus {
    outline: none;
  }
`;
export default Textarea;
