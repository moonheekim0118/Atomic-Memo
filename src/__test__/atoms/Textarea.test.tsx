import 'jsdom-global/register';
import React from 'react';
import Textarea from '../../components/atoms/Textarea';
import { mount } from 'enzyme';

describe('<Textarea/>', () => {
  let container = null;
  const mockFn = jest.fn();

  it('renders correctly', () => {
    container = mount(<Textarea value="" onChange={mockFn} size="title" />);
  });

  it('matches snpashot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('should call onChage if Container is Changed', () => {
    container.simulate('change');
    expect(mockFn).toHaveBeenCalled();
  });
});
