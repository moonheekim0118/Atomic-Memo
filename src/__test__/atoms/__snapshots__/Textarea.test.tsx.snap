// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea/> matches snpashot 1`] = `"<textarea rows=\\"2\\" font-size=\\"2rem\\" placeholder=\\"Title\\" class=\\"Textarea__StyledTextArea-o90l14-0 duiURD\\"></textarea>"`;
